packages/mui-base/src/useMenu/useMenu.ts

```diff
@@ -122,9 +122,9 @@ export function useMenu(parameters: UseMenuParameters = {}): UseMenuReturnValue
   }, [listboxId, registerPopup]);
 
   React.useEffect(() => {
-    if (open && highlightedValue === subitemKeys[0] && !isInitiallyOpen.current) {
+    if (open && highlightedValue && !isInitiallyOpen.current) {
```

Comment:
TBH i don't have context on why `subitemKeys[0]` is explicilty used here, instead of focusing `subitemKeys[0]` focusing `highlightedValue` seems to fix the issue.

Reply:
I was suspicious seeing it as well, and noticed the same issue while working on adding support for selectable menu items. The changed logic looks good to me 👍 cc @michaldudak for the final review.

Reply:
I have to admit I don't remember what was my reasoning here. It may be I was trying some things out and forgot to remove it. Thanks for taking care of it, @sai6855!